### PR TITLE
Create small thumbnails when taking the photo so to save memory

### LINKED ISSUE #49: Insane amounts of RAM usage and non responsive UI
We are loading images which might be 16 MPixels long (depending on the camera) which is around 256 megabytes when its uncompressed from its png format.

### PR DESCRIPTION
**Pull request by vpnda**
Multiple fixes in this commit ( fixes #49 )

- Remove files that were created (previously we only deleted the entry from the db)
- PnR stuff related on how we load images
- Database change: now we store a thumbnail that gets loaded into the recyclerView

RAM usage drops to around 60 MB with 30 images loaded to all of the 3 fragments.
Butter smooth animations on my phone 👍

### CODE REVIEW

File: app/src/main/java/com/example/ubclaunchpad/launchnote/addPhoto/TakePhotoActivity.kt [hunk 1]
```diff
@@ -2,25 +2,31 @@ package com.example.ubclaunchpad.launchnote.addPhoto
...
+import io.reactivex.ObservableEmitter
 import io.reactivex.android.schedulers.AndroidSchedulers
 import io.reactivex.schedulers.Schedulers
-
 import java.io.File
+import java.io.FileOutputStream
 import java.io.IOException
 import java.text.SimpleDateFormat
-import java.util.Date
+import java.util.*
```
Comment (frostyshadows) [inline app/src/main/java/com/example/ubclaunchpad/launchnote/addPhoto/TakePhotoActivity.kt]: You generally don't want to do * imports in Java. [Stackoverflow link](https://stackoverflow.com/questions/147454/why-is-using-a-wild-card-with-a-java-import-statement-bad) if you're interested.

File: app/src/main/java/com/example/ubclaunchpad/launchnote/addPhoto/TakePhotoActivity.kt [hunk 2]
```diff
@@ -62,13 +69,52 @@ class TakePhotoActivity : BaseActivity() {
...
+                .apply(RequestOptions.bitmapTransform(CenterInside()))
+                .apply(RequestOptions().override(maxSize)).submit().get()
+        val f = createImageFile(true)
+        var out: FileOutputStream? = null;
+        try {
+            out = FileOutputStream(f)
+            compressedBmp.compress(Bitmap.CompressFormat.PNG, 100, out)
+        } finally {
+                try {
+                    if (out != null) {
```
Comment (frostyshadows) [inline app/src/main/java/com/example/ubclaunchpad/launchnote/addPhoto/TakePhotoActivity.kt]: You can just write `out?.close()`

File: app/src/main/java/com/example/ubclaunchpad/launchnote/addPhoto/TakePhotoActivity.kt [hunk 3]
```diff
@@ -81,33 +127,37 @@ class TakePhotoActivity : BaseActivity() {
...
+                        onDone.onComplete()
+                    }
         }
-        finish()
-
     }
 
     @Throws(IOException::class)
-    private fun createImageFile(): File {
+    private fun createImageFile(forCompressed: Boolean = false): File {
```
Comment (frostyshadows) [inline app/src/main/java/com/example/ubclaunchpad/launchnote/addPhoto/TakePhotoActivity.kt]: add more documentation/comments

File: app/src/main/java/com/example/ubclaunchpad/launchnote/photoBrowser/AllPhotosAdapter.kt [hunk 4]
```diff
@@ -25,63 +23,65 @@ class AllPhotosAdapter(): RecyclerView.Adapter<AllPhotosAdapter.ViewHolder>() {
     private lateinit var picNotes: List<PicNote>
     private lateinit var context: Context
     private lateinit var picNotesSelected: Set<PicNote>
+    private var scale: Int = 500
+
     var onOnImageActionListener: onImageActionListener? = null;
 
+
```
Comment (frostyshadows) [inline app/src/main/java/com/example/ubclaunchpad/launchnote/photoBrowser/AllPhotosAdapter.kt]: remove extra spaces

**Review by frostyshadows — CHANGES_REQUESTED** [review]
Tested it out and the RAM usage is much better now,
 great work! 👍 Left a few comments.

**Comment by vpnda** [comment]
fixes #47

**Review by frostyshadows — APPROVED** [review]

**Pull request merged**